confmap/provider.go

```diff
@@ -100,21 +102,38 @@ func WithRetrievedClose(closeFunc CloseFunc) RetrievedOption {
 	}
 }
 
-// NewRetrievedFromMap returns a new Retrieved instance that contains a Map data.
-// * conf the Map that will be merged to the given map in the MergeTo.
-// * CloseFunc specifies a function to be invoked when the configuration for which it was
-//   used to retrieve values is no longer in use and should close and release any watchers
-//	 that it may have created.
-func NewRetrievedFromMap(conf *Conf, opts ...RetrievedOption) Retrieved {
+// Deprecated: [v0.53.0] Use NewRetrievedFromYAML, please inform the otel collector maintainers if
+// NewRetrievedFromYAML does not cover all your needs.
+func NewRetrievedFromConf(conf *Conf, opts ...RetrievedOption) Retrieved {
```

Comment:
I think removing this will make our (Datadog's) use case more difficult. In our case, there is no 1:1 mapping between the Datadog Agent's YAML and the underlying Collector configuration and even if there were, we don't see the YAML directly, just a Viper object.

We could do something like build a `confmap.Conf`, marshal it to YAML and pass it to `NewRetrievedFromYAML`, but it would be more cumbersome

Reply:
Was looking for feedback like this. Will not deprecate this, or provide an API that accepts 'map[string] interface {}' is that ok?

Reply:
`map[string]interface{}` would be okay, yes!

Reply:
Done.